content/development/handbook/front-end/react/_index.md

```diff
@@ -5,4 +5,60 @@ tags: ["development", "handbook", "front-end", "react"]
 weight: 100
 ---
 
-TODO
\ No newline at end of file
+### React
+
+React is a library for developing component based applications. It is done by having declared html as xml in javascript code, and that is again transpiled into vanilla javascript for the browser to render into a page.
+
+#### React Components lifecycle
+
+There are sequences that are called when mounting, updated, unmounted and error-catching for every component in React. This is the order they are called:
+
+##### Mounting
+
+1. `constructor()`
+2. `static getDerivedStateFromProps()`
+3. `render()`
+4. `componentDidMount()`
+
+##### Updating
+
+1. `static getDerivedStateFromProps()`
+2. `shouldComponentUpdate()`
+3. `render()`
+4. `getSnapshotBeforeUpdate()`
+5. `componentDidUpdate()`
+
+##### Unmounting
+
+1. `componentWillUnmount()`
+
+##### Error handling
+
+1. `static getDerivedStateFromError()`
+2. `componentDidCatch()`
+
+#### Other React component APIs and properties
+
+##### APIs
+
+1. `setState()`
+2. `forceUpdate()`
+
+##### Class Properties
+
+1. `defaultProps`
+2. `displayName`
+
+##### Instance Properties
+
+1. `props`
+2. `state`
+
+### Adding a new react-app to altinn.studio
+
+When adding a new react-app, use the template folder that is in the github repo. Create a new folder in the `src/react-apps/applications`-folder and copy the contents from the `src/react-apps/templates/template-app`-folder of that newly created folder.  
+Then change the names in the `package.json`-file and start developing the new application.
```

Comment:
Should it be mentioned that you need to comment out run()  from the index.tsx to get redux store to work?

Reply:
You need to comment in not comment out ofcause

Reply:
now that part of the code is commented out, so one needs to comment it in for the redux-saga functionality to work

Reply:
Added docs that says if `run()` is commented out, and the app i going to use sagas, you should uncomment this line of code, since it will initialize the sags.